test/e2e-tests.sh

```diff
@@ -33,6 +33,22 @@ function knative_setup() {
   install_knative_serving
 }
 
+function wait_for_leader_controller() {
+  echo -n "Waiting for a leader Controller"
+  for i in {1..150}; do  # timeout after 5 minutes
+    local leader=$(kubectl get lease controller -n "${SYSTEM_NAMESPACE}" -ojsonpath='{.spec.holderIdentity}' | cut -d"_" -f1)
+    # Make sure the leader pod exists.
```

Comment:
I feel that the pod exists check here is redundant. 
Although `getLeader()` in go code has the existing check, it is because we just want to confirm it after the failover test (the original leader pod is deleted), I guess.

Reply:
Without the check the function doesn't seem complete. There can be cases when the Lease object has stale data so this is making it more robust.

Reply:
I guess it won't harm 🤷‍♂️